File: OCH_Tao/ABC/B/026.py
import math
N = int(input())
l = []
for i in range(N):
  l.append(int(input()))
l.sort(reverse=True)
x = 0
for i in range(N):
  if i%2==0:
    x += l[i]**2
  else:
    x -= l[i]**2
print(x * math.pi)

File: OCH_Tao/ABC/B/026.js
let R = (require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n")).map(x => parseInt(x, 10));
const N = R.shift();
R.sort((a, b) => b - a);
let x = 0;
for (let i = 0; i < N; i++) {
  if (i % 2 === 0) {
    x += R[i] ** 2;
  } else {
    x -= R[i] ** 2;
  }
}
console.log(x * Math.PI);